src/Container/TemplatedErrorHandlerFactory.php

```diff
@@ -38,6 +38,15 @@
  * </code>
  *
  * If any of the keys are missing, default values will be used.
+ *
+ * @deprecated Since 2.0, to remove in 3.0. Please migrate your code to use
```

Comment:
Okay, I came up with the following: the default delegate which I proposed in this patch needs to be a `DelegateInterface`, and should likely return a 404 response. As such, I made the following changes:

- Modified the constructor to allow null, callable, or `DelegateInterface` instances for the default delegate, and validate what I receive, decorating callables with `CallableDelegateWrapper`.
- Created `Zend\Expressive\Delegate\NotFoundDelegate`, which provides a 404 response. I also changed `Zend\Expressive\Middleware\NotFoundHandler` to act as a middleware proxy to the delegate, to allow re-using the logic between the two.
- Modified `getDefaultDelegate()` to pull the service `Zend\Expressive\Delegate\DefaultDelegate` from the container, if the container and the service exist; otherwise, it creates an instance of `NotFoundDelegate` using the current response prototype (but no attempt to retrieve a renderer).
- Updated `run()` to simply call `getDefaultDelegate()`, with no additional logic.
- Updated `ApplicationFactory` to pull the `DefaultDelegate` service if present, using a `null` value otherwise, when creating the `$defaultDelegate` argument for the `Application` instance.

Once those were in place, I removed each of these legacy final handlers. I'll update the deprecations PR for 1.1 to reflect that.

Thanks for the feedback, @xtreamwayz !